### PR TITLE
Snapzone generator

### PR DESCRIPTION
**Pull request by Gusinuhe**
### Description

Snappable properties can now generate Snap Zones.

### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### CODE REVIEW

File: Editor/SnapZone/SnapZoneWizard.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+﻿using UnityEditor;
+using UnityEngine;
+using Innoactive.Creator.XRInteraction;
+
+namespace Innoactive.CreatorEditor.XRInteraction
+{
+    /// <summary>
+    /// Editor Window used to setup settings for <see cref="SnapZone"/>s
```
Comment (Maxsiom) [inline Editor/SnapZone/SnapZoneWizard.cs:8]:
```suggestion
    /// Editor Window used to setup settings for <see cref="SnapZone"/>s.
```

File: Editor/SnapZone/SnapZoneWizard.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            titleContent = new GUIContent("Snap Zone Settings");
+
+            if (settings == null)
+            {
+                window.Close();
+                return;
+            }
+
+            EditorGUILayout.Space();
+            EditorGUILayout.HelpBox("These settings help you to configure Snap Zones within your training. You can define colors and other values that will be set to Snap Zones created by clicking the 'Create Snap Zone' button of a Snappable Property.", MessageType.Info);
```
Comment (Maxsiom) [inline Editor/SnapZone/SnapZoneWizard.cs:44]:
```suggestion
            EditorGUILayout.HelpBox("These settings help you to configure Snap Zones within your scenes. You can define colors and other values that will be set to Snap Zones created by clicking the 'Create Snap Zone' button of a Snappable Property.", MessageType.Info);
```

**Comment by Gusinuhe** [comment]
@tomwim ready for review

File: Runtime/Interaction/SnapZone.cs [hunk 3]
```diff
@@ -34,14 +35,14 @@ public GameObject ShownHighlightObject
...
-
+        
         /// <summary>
         /// The color of the material used to draw the <see cref="ShownHighlightObject"/>.
         /// Use the alpha value to specify the degree of transparency.
         /// </summary>
         public Color ShownHighlightObjectColor
         {
-            get { return shownHighlightObjectColor; }
+            get => shownHighlightObjectColor;
```
Comment (aleksei-korolev) [inline Runtime/Interaction/SnapZone.cs:45]: Inconsistent code style

File: Editor/SnapZone/SnapZoneSettings.cs [hunk 4]
```diff
@@ -0,0 +1,158 @@
+using System.IO;
+using UnityEditor;
+using UnityEngine;
+using Innoactive.Creator.XRInteraction;
+
+namespace Innoactive.CreatorEditor.XRInteraction
+{
+    /// <summary>
+    /// Settings for <see cref="SnapDropZone"/>s for e.g. automatic creation of such snap zones.
```
Comment (tomwim) [inline Editor/SnapZone/SnapZoneSettings.cs:9]:
```suggestion
    /// Settings for <see cref="SnapZone"/>s for e.g. automatic creation of such snap zones.
```

File: Editor/SnapZone/SnapZoneSettings.cs [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+        private Material validationMaterial;
+
+        /// <summary>
+        /// The material used for drawing when an <see cref="InteractableObject"/> is hovering a <see cref="SnapZone"/>. Should be transparent.
+        /// </summary>
+        public Material HighlightMaterial
+        {
+            get
+            {
+                if (highlightMaterial == null)
```
Comment (aleksei-korolev) [inline Editor/SnapZone/SnapZoneSettings.cs:47]: You can move the body to SetupHighlightMaterial() method to make it more readable.

File: Editor/SnapZone/SnapZoneSettings.cs [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        /// <summary>
+        /// The material used for the highlight object. Should be transparent.
+        /// </summary>
+        public Material ValidationMaterial
+        {
+            get
+            {
+                if (validationMaterial == null)
+                {
+                    validationMaterial = CreateMaterial();
```
Comment (aleksei-korolev) [inline Editor/SnapZone/SnapZoneSettings.cs:75]: same

File: Editor/SnapZone/SnapZoneSettings.cs [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+        /// Loads the first existing <see cref="SnapZoneSettings"/> found in the project.
+        /// If non are found, it creates and saves a new instance with default values.
+        /// </summary>
+        public static SnapZoneSettings Settings
+        {
+            get
+            {
+                if (settings == null)
+                {
+                    string filter = "t:ScriptableObject SnapZoneSettings";
```
Comment (aleksei-korolev) [inline Editor/SnapZone/SnapZoneSettings.cs:106]: same

**Pull request merged**